### ISSUE TITLE
test(vcs): JitDiffReport absence contract is unasserted below the top level

### ISSUE THREAD

**Issue by dekobon** [dekobon/big-code-analysis#1290]
## Summary

`JitDiffReport`'s "unavailable groups are absent, not zero" contract is asserted only at the top level of the report. The `contributions` sub-object — where the hazard the design guards against actually lives — is never inspected by any test, in any of the three suites that claim to cover it.

Found by an `audit-tests` pass during the 2026-08-08 quick-fix batch (evidence is source inspection, not a perturbation run — the tree had a validation gate running).

## The contract

`src/vcs/jit.rs:374-385` defines `JitDiffContributions` (size + diffusion only), and its own doc states the point explicitly:

> unlike `JitContributions`, there is no zero-valued field a consumer could misread as "this group is low risk"

`JitContributions` (`src/vcs/jit.rs:192-203`) carries `history`, `purpose`, `experience` as `f64`.

## The gap

Three tests assert absence, all with the same top-level-only loop:

- `big-code-analysis-py/tests/test_vcs.py:536-537`
- `big-code-analysis-cli/tests/vcs/vcs_jit.rs:310-317`
- `tests/vcs/vcs_jit.rs:395-441`

Each checks `"history" not in report`, `"experience" not in report`, and so on. That genuinely catches "diff mode accidentally emitted a full commit report", because `commit` and `risk_score` are top-level fields.

It does not catch the failure the struct doc warns about. If `JitDiffReport.contributions` were retyped to `JitContributions` — the obvious "just reuse the commit type" simplification — the serialized report would carry `contributions.history = 0.0`, `contributions.experience = 0.0`, `contributions.purpose = 0.0`, and **all three tests would still pass**, because those keys are nested one level down. `rg contributions big-code-analysis-py/tests/*.py` returns nothing: no Python test reads `report["contributions"]` at all.

## Suggested fix

One assertion per suite, pinning the key set rather than probing individual names:

```python
assert set(report["contributions"]) == {"size", "diffusion"}
```

Pinning the set (not `"history" not in report["contributions"]`) is what makes it a guard: a future group added to the diff path fails loudly instead of slipping through an absence check nobody extended.

## Secondary, same file

`test_vcs_jit_commit_returns_report` (`test_vcs.py:501-502`) asserts `isinstance(report["features"][group], dict)`, which an empty `{}` satisfies — a regression emitting the group keys with no fields inside would pass. `assert isinstance(report["risk_score"], (int, float))` (`:498`) also accepts `True`, since `bool` subclasses `int` in Python, and pins no range. The Rust sibling is stronger (`big-code-analysis-cli/tests/vcs/vcs_jit.rs:81` asserts `.is_number()`).

## Impact

Test-quality only; no shipped behaviour is wrong today. Filed rather than fixed inline because it is pre-existing and shared across all three suites, so it is a coherent change of its own rather than a rider on a documentation batch.

---

## Resolution Plan

Confirmed on `main` (`d65b3049`). `JitDiffContributions`
(`src/vcs/jit.rs:380`) carries exactly `size` + `diffusion`;
`diff_mode_marks_unavailable_groups_distinct_from_zero`
(`tests/vcs/vcs_jit.rs:395`) loops six **top-level** keys and never
opens `contributions`; `rg contributions
big-code-analysis-py/tests/*.py` returns nothing.

1. **Add the key-set assertion to all three suites**, pinning the set
   rather than probing names — `assert set(report["contributions"]) ==
   {"size", "diffusion"}` in `test_vcs.py`, and the `serde_json::Value`
   equivalent in `big-code-analysis-cli/tests/vcs/vcs_jit.rs` and
   `tests/vcs/vcs_jit.rs`. Set equality is what makes it a guard: a
   future group added to the diff path fails loudly instead of slipping
   past an absence check nobody extended.
2. **Prove the new assertion discriminates before believing it.** The
   perturbation is the one the struct doc names: retype
   `JitDiffReport.contributions` to `JitContributions` and confirm the
   three new assertions are the failures — and that the six existing
   top-level absence checks still pass, which is the evidence that the
   old test genuinely could not see this. Per `.claude/rules/testing.md`,
   restore by inverse edit, not `git checkout -- <file>`.
3. **Keep the existing top-level loop.** It covers a different
   regression ("diff mode emitted a full commit report") and the two
   assertions are not redundant. Widening the loop to walk nested
   objects would blur both claims into one; add the second assertion
   beside it instead.
4. **Fix the two secondary weaknesses in `test_vcs.py` while there.**
   `isinstance(report["features"][group], dict)` (`:501-502`) is
   satisfied by `{}` — assert the expected key set per group instead.
   `isinstance(report["risk_score"], (int, float))` (`:498`) accepts
   `True`, since `bool` subclasses `int` in Python — assert
   `not isinstance(..., bool)` and a range, matching the stronger Rust
   sibling at `big-code-analysis-cli/tests/vcs/vcs_jit.rs:81`.
5. **No production change, so no snapshot or baseline churn.** Run the
   full workspace suite plus `make py-test`; the Python half is the one
   `cargo nextest` does not cover.

## Assessment

| Dimension  | Rating |
|------------|--------|
| Difficulty | Low |
| Complexity | Low |
| Priority   | Low |

**Comment by dekobon** [comment]
Added resolution plan and assessment ratings (difficulty Low / complexity Low / priority Low; `low-priority` was already applied). Confirmed the gap: `JitDiffContributions` (src/vcs/jit.rs:380) carries only `size`/`diffusion`, the Rust test at tests/vcs/vcs_jit.rs:395 loops six top-level keys, and `rg contributions big-code-analysis-py/tests/*.py` returns nothing. The plan adds a perturbation step — retype `contributions` to `JitContributions` and confirm the new assertions are the only failures — since a passing set-equality assertion alone would not show the old test could not see this.
